x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -124,6 +164,17 @@
     public static final String RULE_FIELD_NAME = "rule";
     public static final String OPAQUE_ID_FIELD_NAME = "opaque_id";
     public static final String X_FORWARDED_FOR_FIELD_NAME = "x_forwarded_for";
+    // the fields below are used exclusively for "security_configuration_change" type of events, and show the configuration
+    // object taking effect; it could be creating a new, or updating an existing configuration
+    // if our (REST) APIs (at least the security APIs) would make the distinction between creating a *new* resource using the POST
+    // verb and updating an *existing* resource using the PUT verb, then auditing would also be able to show the create/update distinction
+    public static final String PUT_CONFIG_FIELD_NAME = "put";
+    public static final String DELETE_CONFIG_FIELD_NAME = "delete";
+    public static final String CHANGE_PASSWORD_FIELD_NAME = "change_password";
+    public static final String ENABLE_CONFIG_FIELD_NAME = "enable";
+    public static final String DISABLE_CONFIG_FIELD_NAME = "disable";
+    public static final String CREATE_API_KEY_FIELD_NAME = "create.apikey";
+    public static final String INVALIDATE_API_KEYS_FIELD_NAME = "invalidate_apikeys";
```

Comment:
The name `put` looks a bit off in this context. I understand it is a carry over from the Rest action. But `put` or `post` is too HTTP specific and do not seem to blend in very well with other verbs like `create`, `enable` etc. I'd suggest to use `upsert`.

I also wonder whether we could have more consistency and simplicity in the names. The mixed usage of underscore and dot looks a bit standing out. Can we use just either of them, but not both? Some of the fields are named as `verb.noun` or `verb_noun`, is it possible to make these field more generic to contain only the verb. The `noun` part of it will be in the body anyway.

Names are hard. These are just my 2cents.

Reply:
> The name put looks a bit off in this context. I understand it is a carry over from the Rest action. But put or post is too HTTP specific and do not seem to blend in very well with other verbs like create, enable etc. I'd suggest to use upsert.

It is not a carry over. I gave it careful thinking. I think `put`, by it's analogy with the HTTP verb, carries the meaning that this is "the new configuration taking effect". No prob to use `upsert` if you like that more.

> I also wonder whether we could have more consistency and simplicity in the names. The mixed usage of underscore and dot looks a bit standing out. Can we use just either of them, but not both? Some of the fields are named as verb.noun or verb_noun, is it possible to make these field more generic to contain only the verb. The noun part of it will be in the body anyway.

I think we have to use both, but I'll take any specific suggestions you've got, we can brainstorm the naming if you're up to. Let me grab an exhaustive list of examples and we can discuss more to the point.